hs/src/Reach/Eval/Core.hs

```diff
@@ -4440,34 +4441,83 @@ evalId_ lab x = do
 evalId :: String -> SLVar -> App SLSVal
 evalId lab x = sss_sls <$> evalId_ lab x
 
+parseCoefficient :: String -> EvalError -> App (Rational, Maybe Int)
+parseCoefficient b err = do
+  let parseInt i =
+        (fromIntegral iInt, precision)
+        where precision = Nothing
+              iInt = read i :: Integer
```

Comment:
I think, but I'm not sure, that these `read`s can be `readMaybe`s and then we can generate errors or impossibles. But I could be wrong